lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,367 @@
+    description:
+      - the mountpoint on which to apply the quotas
+    required: true
+  bhard:
+    description:
+      - hard blocks quota limit
+  bsoft:
+    description:
+      - soft blocks quota limit
+  ihard:
+    description:
+      - hard inodes quota limit
+  isoft:
+    description:
+      - soft inodes quota limit
+  rtbhard:
+    description:
+      - hard realtime blocks quota limit
+  rtbsoft:
+    description:
+      - soft realtime blocks quota limit
+  state:
+    description:
+      - Whether to apply the limits or remove them.
+      - When removing limit, they are set to 0, and not quite removed.
+    default: present
+    choices:
+      - present
+      - absent
+
+notes:
+   - Not tested on any debian based system
+requirements:
+   - xfsprogs
+'''
+
+EXAMPLES = r'''
+- name: Set default project soft and hard limit on /opt of 1g
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    bsoft: 1g
+    bhard: 1g
+    state: present
+
+- name: Remove the default limits on /opt
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    state: absent
+
+- name: Set default soft user inode limits on /home of 1024 inodes and hard of 2048
+  xfs_quota:
+    type: user
+    mountpoint: /home
+    isoft: 1024
+    ihard: 2048
+
+'''
+
+RETURN = ''' # '''
+
+import os
+import sys
+import json
+import pwd
+import grp
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import human_to_bytes
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            type=dict(required=True, choices=['user', 'group', 'project']),
+            name=dict(required=False, default=None),
+            mountpoint=dict(required=True),
+            bhard=dict(required=False, default=None),
+            bsoft=dict(required=False, default=None),
+            ihard=dict(required=False, default=None),
+            isoft=dict(required=False, default=None),
+            rtbhard=dict(required=False, default=None),
+            rtbsoft=dict(required=False, default=None),
+            state=dict(required=False, default='present', choices=['present', 'absent'])
+        ),
+        supports_check_mode=True
+    )
+
+    quota_type = module.params['type']
+    name = module.params['name']
+    mountpoint = module.params['mountpoint']
+    bhard = module.params['bhard']
+    bsoft = module.params['bsoft']
+    ihard = module.params['ihard']
+    isoft = module.params['isoft']
+    rtbhard = module.params['rtbhard']
+    rtbsoft = module.params['rtbsoft']
+    state = module.params['state']
+
+    if bhard is not None:
+        bhard = human_to_bytes(bhard)
+
+    if bsoft is not None:
+        bsoft = human_to_bytes(bsoft)
+
+    if rtbhard is not None:
+        rtbhard = human_to_bytes(rtbhard)
+
+    if rtbsoft is not None:
+        rtbsoft = human_to_bytes(rtbsoft)
+
+    changed = False
+
+    if os.getuid() != 0:
+        module.fail_json(msg='You need to be root to run this module')
```

Comment:
This then would become:
```suggestion
        module.fail_json(msg='You need to be root to run this module', **result)
```

Reply:
Also, I am not sure if this is correct, do you need to be root, or do you need to have the right capabilities ?

Reply:
Good question. I'd need to investigate.

Reply:
CAP_SYS_ADMIN seems to be correct.
So I just drop the 'check if root' part?

Reply:
So since it is not that easy to test if you have CAP_SYS_ADMIN, it is likely easier to try to do it, and if it fails you likely do not have the capabilities. (Not sure if you can catch why it failed, and give different errors wrt. the failure). If not, you could get away with a basic error message (Failed because you may not be root or have CAP_SYS_ADMIN capabilities, or another reason...).

Reply:
done